### PR TITLE
ignore blank strings from history.add

### PR DESCRIPTION
**Pull request by thewisenerd**

### CODE REVIEW

File: Maccy/History.swift [hunk 1]
```diff
@@ -31,7 +31,25 @@ class History {
     return savedHistory
   }
 
+  func isBlank(_ string: String) -> Bool {
+    if string.count == 0 {
+      return true
+    }
+
+    if let first = string.first {
+      if first == "\n" {
```
Comment (p0deje) [inline Maccy/History.swift]: This will return true for `\n something`. I guess that's not the intention.

File: Maccy/History.swift [hunk 2]
```diff
@@ -31,7 +31,25 @@ class History {
     return savedHistory
   }
 
+  func isBlank(_ string: String) -> Bool {
```
Comment (p0deje) [inline Maccy/History.swift]: There is actually no need to have a custom function - you can use Swift for that:

```swift
string.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty
```

Here are examples of how it works:

```
(lldb) po "foo".trimmingCharacters(in: .whitespacesAndNewlines).isEmpty
false

(lldb) po "".trimmingCharacters(in: .whitespacesAndNewlines).isEmpty
true

(lldb) po " ".trimmingCharacters(in: .whitespacesAndNewlines).isEmpty
true

(lldb) po "\n".trimmingCharacters(in: .whitespacesAndNewlines).isEmpty
true

(lldb) po "\n ".trimmingCharacters(in: .whitespacesAndNewlines).isEmpty
true

(lldb) po "\n foo".trimmingCharacters(in: .whitespacesAndNewlines).isEmpty
false

(lldb) po " foo".trimmingCharacters(in: .whitespacesAndNewlines).isEmpty
false
```

File: Maccy/History.swift [hunk 3]
```diff
@@ -31,7 +31,25 @@ class History {
...
+        return true
+      }
+    }
+
+    return false
+  }
+
   func add(_ string: String) {
+    if isBlank(string) {
+        return
```
Comment (p0deje) [inline Maccy/History.swift]: Please use 2 spaces indentation.

**Comment by thewisenerd** [comment]
resolved

**Pull request merged**